src/main.cpp

```diff
@@ -264,4 +273,86 @@ void runSteeringDisplay() {
 	while (1) {
 		display.run();
 	}
+}
+
+void buttonRead(int buttonPress){
+	int ledPos;
+	for(int i = 0; i <8; i++){
+		shiftClk = 1;
+		wait_us(20);
+		shiftClk = 0;
+		shiftLatch = 1;
+
+		if(i == 3 && buttonPress == 1){
+			ledPos = 1;
+			ledOn(ledPos);
+		}
+
+		else if(i == 4 && buttonPress == 1){
+			ledPos = 2;
+			ledOn(ledPos);
+		}
+
+		else if(i == 5 && buttonPress == 1){
+			ledPos = 3;
+			ledOn(ledPos);
+		}
+
+		else if(i == 6 && buttonPress == 0){
+			ledPos = 4;
+			ledOn(ledPos);
+		}
+
+		else if(i == 6 && buttonPress == 1){
+			ledPos = 5;
+			ledOn(ledPos);
+		}
+
+		else if(i == 7 && buttonPress == 1){
+			ledPos = 6;
+			ledOn(ledPos);
+		}
+
+		wait_us(20);
+		shiftLatch = 0;
+	}
+}
+
+void ledOn(int ledPos){
+	for(int i = 0; i < 8; i++){
+		shiftClk = 1;
+		wait_us(20);
+		shiftClk = 0;
+		shiftLatch = 1;
```

Comment:
For the output register, the latch moves the data from the shift register to the storage register. See this gif: https://lastminuteengineers.b-cdn.net/wp-content/uploads/arduino/74HC595-Shift-Register-Working.gif

So I think you need to set the latch high and low after the loop is over.

Reply:
Oh okay, so I'll set it high and then set it low after a delay